guppylang/prelude/_internal/compiler/array.py

```diff
@@ -207,3 +225,30 @@ def compile_with_inouts(self, args: list[Wire]) -> CallReturnWires:
 
     def compile(self, args: list[Wire]) -> list[Wire]:
         raise InternalGuppyError("Call compile_with_inouts instead")
+
+
+class ArrayIterEndCompiler(ArrayCompiler):
+    """Compiler for the `ArrayIter.__end__` method."""
+
+    def compile(self, args: list[Wire]) -> list[Wire]:
+        # For linear array iterators, we need to make sure that all array elements have
+        # been taken out (i.e. replaced by `None`)
+        if self.elem_ty.type_bound() == ht.TypeBound.Any:
```

Comment:
Isn't there something like `self.elem_ty.is_linear()` you can use here?

Reply:
I don't think that exists: https://cqcl.github.io/hugr/generated/hugr.tys.Type.html

Reply:
Oh this is not guppy-specific anymore, right